### PR TITLE
[DPM-63] signidice processor and session updates

### PR DESCRIPTION
**Pull request by algys**

### CODE REVIEW

File: server/app.go [hunk 1]
```diff
@@ -255,6 +254,18 @@ func startAmc(a *App, ctx context.Context) error {
 		if ok, err := listener.Subscribe(1, 0); err != nil || !ok {
 			log.Fatal().Msgf("Action monitor subscribe error: %v", err)
 		}
+		if ok, err := listener.Subscribe(2, 0); err != nil || !ok {
```
Comment (justefg) [inline server/app.go:257]: use for cycle, please :)

File: signidice/usecase/usecase.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+
+func (a *SignidiceUseCase) rsaSign(digest eos.Checksum256) (string, error) {
+	//hash := sha256.New()
+	//hash.Write(digest)
+
+	sign, err := rsa.SignPKCS1v15(rand.Reader, a.rsaKey, crypto.SHA256, digest)
+	if err != nil {
+		return "", err
```
Comment (justefg) [inline signidice/usecase/usecase.go:54]: don't use tabs

File: game_sessions/usecase/usecase.go [hunk 3]
```diff
@@ -105,6 +112,77 @@ func (a *GameSessionsUseCase) NewSession(ctx context.Context, Casino *models.Cas
...
+	if err != nil {
+		return err
+	}
+
+	game, err := a.casinoRepo.GetGame(ctx, gs.GameID)
+	if err != nil {
+		return err
+	}
+
+	bcAction := eos.Action{
```
Comment (justefg) [inline game_sessions/usecase/usecase.go:127]: & here would be better for code readability, I think

File: game_sessions/usecase/usecase.go [hunk 4]
```diff
@@ -105,6 +112,77 @@ func (a *GameSessionsUseCase) NewSession(ctx context.Context, Casino *models.Cas
...
+			ActionType   uint16   `json:"type"`
+			ActionParams []uint32 `json:"params"`
+		}{
+			SessionId:    gs.BlockchainSesID,
+			ActionType:   actionType,
+			ActionParams: actionParams,
+		}),
+	}
+
+	trxOpts := eos.TxOptions{}
```
Comment (justefg) [inline game_sessions/usecase/usecase.go:145]: & for same reasons

File: eventprocessor/processor.go [hunk 5]
```diff
@@ -21,34 +23,41 @@ const (
...
+	gsRepo := p.repos.GameSession
+
 	bcSession, err := gsRepo.GetSessionByBlockChainID(ctx, event.RequestID)
 	if err != nil {
 		log.Warn().Msgf("Couldn't find session with requestID %v", event.RequestID)
 		return
 	}
+
+	// already processed offset
+	if bcSession.LastOffset >= event.Offset {
```
Comment (justefg) [inline eventprocessor/processor.go:52]: log debug

File: signidice/usecase/usecase.go [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+	return &SignidiceUseCase{
+		bc:                  bc,
+		rsaKey:              key,
+		platformAccountName: platformAccountName,
+	}
+}
+
+
+func (a *SignidiceUseCase) rsaSign(digest eos.Checksum256) (string, error) {
+	//hash := sha256.New()
```
Comment (justefg) [inline signidice/usecase/usecase.go:49]: remove comments

File: server/app.go (hunk 1 above)
Comment (algys) [reply server/app.go]: fixed

File: game_sessions/usecase/usecase.go (hunk 3 above)
Comment (algys) [reply game_sessions/usecase/usecase.go]: ok

File: game_sessions/usecase/usecase.go (hunk 4 above)
Comment (algys) [reply game_sessions/usecase/usecase.go]: ok

File: eventprocessor/processor.go (hunk 5 above)
Comment (algys) [reply eventprocessor/processor.go:52]: done

File: signidice/usecase/usecase.go (hunk 6 above)
Comment (algys) [reply signidice/usecase/usecase.go]: oops..., fixed

File: signidice/usecase/usecase.go (hunk 2 above)
Comment (algys) [reply signidice/usecase/usecase.go:51]: so why ? :)

**Pull request merged**